### PR TITLE
Improvements for the OS parser a second level of regular expressions has been added

### PR DESCRIPTION
**Pull request by sanchezzzhak**
### Description:

+ moved 102 Apple regular expressions to a regular expression subarray
+ the repetition of constructs when determining the iOS or MAC version has also been reduced

### Review

* [ ] Functional review done
* [ ] Potential edge cases thought about (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] Usability review done (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] Security review done [see checklist](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] Code review done
* [ ] Tests were added if useful/possible
* [ ] Reviewed for breaking changes
* [ ] Developer changelog updated if needed
* [ ] Documentation added if needed
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: regexes/oss.yml
Comment (sgiehl) [inline regexes/oss.yml:880]: Not sure if we should combine them. This might cause that the OS is returned in different writing cases, depending how it was written in the user agent.

Comment (sanchezzzhak) [reply regexes/oss.yml]: returned the changes back

**Comment by sanchezzzhak** [comment]
I accidentally closed this PR

File: regexes/oss.yml
Comment (sgiehl) [inline regexes/oss.yml:301]: I would suggest to rename this one into `versions` instead of `regexs` and maybe either have `version:` or `versions:`

**Review by sgiehl — APPROVED** [review]

**Comment by sgiehl** [comment]
@sanchezzzhak Feel free to merge this one if you think it's fine. The next version after this one being merge will be a major release, so the ports to other languages might notice the change and update their code.

**Pull request merged**
